File: frontend/src/lib/api/public/types/children.ts
import { z } from 'zod';

import { medicalReviewStatusSchema } from './enums';


// Children Filters

export const listChildrenFilterRequestSchema = z.object({
	is_active: z.boolean().nullable().optional(),
	review_status: medicalReviewStatusSchema
		.nullable()
		.optional()
});

export type ListChildrenFilterRequest =
	z.infer<typeof listChildrenFilterRequestSchema>;


// Child Response

export const childResponseSchema = z.object({
	id: z.uuid(),
	location_id: z.uuid(),
	user_id: z.uuid(),

	first_name: z.string(),
	last_name: z.string(),
	dob: z.string(),

	medical_info: z.string().nullable(),
	allergy_info: z.string().nullable(),
	medication_info: z.string().nullable(),

	is_active: z.boolean(),
	review_status: medicalReviewStatusSchema,

	created_at: z.string(),
	updated_at: z.string()
});

export type ChildResponse =
	z.infer<typeof childResponseSchema>;


// Child Table Response

export const childTableResponseSchema = z.object({
	id: z.uuid(),
	location_id: z.uuid(),
	user_id: z.uuid(),

	first_name: z.string(),
	last_name: z.string(),
	dob: z.string(),

	is_active: z.boolean(),
	review_status: medicalReviewStatusSchema
});

export type ChildTableResponse =
	z.infer<typeof childTableResponseSchema>;


// Child Create Request

export const createChildRequestSchema = z.object({
	first_name: z.string(),
	last_name: z.string(),
	dob: z.date(),
	medical_info: z.string().nullable(),
	allergy_info: z.string().nullable(),
	medication_info: z.string().nullable()
});

export type CreateChildRequest =
	z.infer<typeof createChildRequestSchema>;

File: backend/app/public/schemas/children.py
from datetime import date, datetime
from uuid import UUID

from pydantic import BaseModel

from app.public.models.enums import MedicalReviewStatus


class ChildResponse(BaseModel):
    id: UUID
    location_id: UUID
    user_id: UUID

    first_name: str
    last_name: str
    dob: date

    medical_info: str | None
    allergy_info: str | None
    medication_info: str | None

    is_active: bool
    review_status: MedicalReviewStatus

    created_at: datetime
    updated_at: datetime


class ChildTableResponse(BaseModel):
    id: UUID
    location_id: UUID
    user_id: UUID

    first_name: str
    last_name: str
    dob: date

    is_active: bool
    review_status: MedicalReviewStatus


class ListChildrenFilterRequest(BaseModel):
    is_active: bool | None = None
    review_status: list[str] | None = None


class CreateChildRequest(BaseModel):
    first_name: str
    last_name: str
    dob: date

    medical_info: str | None
    allergy_info: str | None
    medication_info: str | None